File: src/ssl/qsslconfiguration.cpp
#include "cxx-qt-io/qsslconfiguration.h"

#include <cxx-qt-io/assertion_utils.h>
#include <string_view>

assert_shared_pointer_type(QSslConfiguration);

namespace rust {
namespace cxxqtio1 {
::rust::String
qsslconfigurationALPNProtocolHTTP2()
{
  return QSslConfiguration::ALPNProtocolHTTP2;
}

::rust::String
qsslconfigurationNextProtocolHttp1_1()
{
  return QSslConfiguration::NextProtocolHttp1_1;
}

}
}

File: tests/standalone/rust/src/qsslconfiguration.rs
use cxx_qt_io::QSslConfiguration;

#[cxx::bridge]
mod ffi {
    #[namespace = "ffi::qsslconfiguration"]
    extern "Rust" {
        fn alpn_protocol() -> &'static str;
        fn next_protocol() -> &'static str;
    }
}

fn alpn_protocol() -> &'static str {
    QSslConfiguration::ALPNProtocolHTTP2
}

fn next_protocol() -> &'static str {
    QSslConfiguration::NextProtocolHttp1_1
}
